### PR TITLE
CLOUDP-187014: Make atlas quickstart command an alias for atlas setup command

### PR DESCRIPTION
**Pull request by VincentPikand**
## Proposed changes

This PR merges `atlas quickstart` and `atlas setup`. 
The philosophy in merging was to have as little change as possible in how the code works to:
1. avoid edge case failures 
2. make reviewing the code a bit easier, as reviewing refactoring + merging is unfeasible

_Please check out the JIRA ticket for manual testing flows._

_Jira ticket:_ CLOUDP-187014

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by kyuan-mongodb — APPROVED** [review]
LGTM!

File: internal/cli/atlas/setup/setup_cmd.go [hunk 1]
```diff
@@ -48,14 +108,281 @@ type ProfileReader interface {
...
+	Provider            string
+	Region              string
+	IPAddresses         []string
+	IPAddressesResponse string
+	DBUsername          string
+	DBUserPassword      string
+	SampleDataJobID     string
+	Tier                string
+	Tag                 map[string]string
+
```
Comment (tibulca) [inline internal/cli/atlas/setup/setup_cmd.go:127]:
```suggestion
```

File: internal/cli/atlas/setup/setup_cmd.go [hunk 2]
```diff
@@ -48,14 +108,281 @@ type ProfileReader interface {
...
+	SkipSampleData              bool
+	SkipMongosh                 bool
+	DefaultValue                bool
+	Confirm                     bool
+	CurrentIP                   bool
+	EnableTerminationProtection bool
+
```
Comment (tibulca) [inline internal/cli/atlas/setup/setup_cmd.go:134]:
```suggestion
```

File: internal/cli/atlas/setup/setup_cmd.go [hunk 3]
```diff
@@ -48,14 +108,281 @@ type ProfileReader interface {
...
+	flags   *pflag.FlagSet
+	flagSet map[string]struct{}
+
 	// control
 	skipRegister bool
 	skipLogin    bool
 }
 
+type setup struct {
```
Comment (tibulca) [inline internal/cli/atlas/setup/setup_cmd.go:143]:
```suggestion
type clusterSettings struct {
```
[nit] I know this was migrated from quickstart (where it was called `quickstart`) but might be good to have a name that reflects the purpose of this struct

File: internal/cli/atlas/setup/setup_cmd.go [hunk 4]
```diff
@@ -77,15 +404,102 @@ This command will help you
...
+
+	return opts.PreRunE(
+		opts.initStore(ctx),
+		opts.register.SyncWithOAuthAccessProfile(defaultProfile),
+		opts.register.InitFlow(defaultProfile),
+		opts.InitOutput(outWriter, ""),
+	)
+}
+
+func (opts *Opts) clusterRun() error {
```
Comment (tibulca) [inline internal/cli/atlas/setup/setup_cmd.go:429]:
```suggestion
func (opts *Opts) setupCluster() error {
```

**Review by tibulca — APPROVED** [review]
LGTM, [e2e were successful](https://spruce.mongodb.com/task/mongodb_atlas_cli_master_e2e_interactive_atlas_interactive_e2e_1d731e0594937eceb6fb0bdf286672b13763cbec_23_07_11_10_02_09/tests?execution=0&sortBy=STATUS&sortDir=ASC), also run positive scenarios on local env for setup (new or existing account) and quickstart (existing account)

**Review by tibulca — APPROVED** [review]

**Review by tibulca — APPROVED** [review]

**Pull request merged**